frontend/public/components/modals/configure-cluster-upstream-modal.tsx

```diff
@@ -0,0 +1,129 @@
+import * as React from 'react';
+
+import { ClusterVersionModel } from '../../models';
+import { ClusterVersionKind, k8sPatch } from '../../module/k8s';
+import {
+  ModalBody,
+  ModalComponentProps,
+  ModalSubmitFooter,
+  ModalTitle,
+  createModalLauncher,
+} from '../factory/modal';
+import { ExternalLink, HandlePromiseProps, withHandlePromise } from '../utils';
+import { useTranslation } from 'react-i18next';
+import { TFunction } from 'i18next';
+import { RadioInput } from '../radio';
+import { CLUSTER_VERSION_DEFAULT_UPSTREAM_SERVER_URL_PLACEHOLDER } from '@console/shared/src/constants';
+import { TextInput } from '@patternfly/react-core';
+import { openshiftHelpBase } from '@console/internal/components/utils';
+
+export const ConfigureClusterUpstreamModal = withHandlePromise(
+  (props: ConfigureClusterUpstreamModalProps) => {
+    const { cv, handlePromise, close } = props;
+    const currentUpstream = cv?.spec?.upstream;
+
+    const [customSelected, setCustomSelected] = React.useState(!!currentUpstream);
+    const [customUrl, setCustomURL] = React.useState(currentUpstream);
+    const [invalidCustomUrl, setInvalidCustomURL] = React.useState(false);
```

Comment:
nit: inconsistent case Url vs URL in these variables

Reply:
Done. Converted to uppercase URL